src/service_daemon.rs

```diff
@@ -299,12 +301,12 @@ impl ServiceDaemon {
         }
 
         // Add mDNS sockets to the poller.
-        for (ipv4, if_sock) in zc.intf_socks.iter() {
-            // It is OK to convert to `usize` here as we only support 32-bit
-            // or 64-bit platforms.
-            let key = u32::from(*ipv4) as usize;
+        for (ip, if_sock) in zc.intf_socks.iter() {
+            let key = zc.id_count;
+            zc.id_count += 1;
+            let _ = zc.poll_ids.insert(key, *ip);
```

Comment:
Nice! 

I think you might be able to go even further, i.e. combine these 3 lines into a method of `zc`:  for example:

`let key = zc.add_poll(*ip)`  

where `add_poll` does the above, and returns the `key`.

Reply:
I had a little fight with the borrow checker for this one, I ended with something a bit weird, maybe there is a better way.

Reply:
Indeed the borrow checker wouldn't like the simple conversion I mentioned. Thanks for the update.